### PR TITLE
Add tests, README, examples

### PR DESCRIPTION
**Pull request by sgfn**
Fixtures are from big buck bunny

### CODE REVIEW

File: test/decoder_native_test.exs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    assert {:ok, file} = File.read(in_path)
+    assert {:ok, decoder_ref} = Native.create()
+    assert <<frame::bytes-size(8000), _rest::binary>> = file
+    assert {:ok, [frame]} = Native.decode(frame, decoder_ref)
+    assert {:ok, []} = Native.flush(decoder_ref)
+    assert Payload.size(frame) == 16_000
+    assert {:ok, ref_file} = File.read(ref_path)
+    assert <<ref_frame::bytes-size(16_000), _rest::binary>> = ref_file
+    assert Payload.to_binary(frame) == ref_frame
+  end
```
Comment (Rados13) [inline test/decoder_native_test.exs]: Why do we decode only 8000 bytes instead of the whole file?

File: test/decoder_test.exs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        child(:file_src, %Membrane.File.Source{chunk_size: 40_960, location: in_path})
+        |> child(:decoder, Decoder)
+        |> child(:sink, %Membrane.File.Sink{location: out_path})
+    )
+  end
+
+  defp assert_files_equal(file_a, file_b) do
+    assert {:ok, a} = File.read(file_a)
+    assert {:ok, b} = File.read(file_b)
+    assert a == b
```
Comment (Rados13) [inline test/decoder_test.exs:33]: I think we could first check that the length is the same and then that they are equal.

File: test/encoder_native_test.exs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    assert {:ok, file} = File.read(in_path)
+    assert {:ok, encoder_ref} = Native.create(:s16le)
+    assert <<frame::bytes-size(16_000), _rest::binary>> = file
+    assert {:ok, [frame]} = Native.encode(frame, encoder_ref)
+    assert {:ok, []} = Native.flush(encoder_ref)
+    assert Payload.size(frame) == 8_000
+    assert {:ok, ref_file} = File.read(ref_path)
+    assert <<ref_frame::bytes-size(8_000), _rest::binary>> = ref_file
+    assert Payload.to_binary(frame) == ref_frame
+  end
```
Comment (Rados13) [inline test/encoder_native_test.exs]: Same issue as in decoder.

**Review by Rados13 — COMMENTED** [review]
As we discussed, a usage example would be beneficial, and its compilation should be checked in CI.

File: test/decoder_native_test.exs (hunk 1 above)
Comment (sgfn) [reply test/decoder_native_test.exs]: changed

File: .circleci/config.yml [hunk 4]
```diff
@@ -2,6 +2,23 @@ version: 2.1
 orbs:
   elixir: membraneframework/elixir@1
 
+executors:
+  examples-executor:
+    resource_class: large
```
Comment (Rados13) [inline .circleci/config.yml]: Do we need an executor with class large for the example of one of the simplest audio codecs?

File: README.md [hunk 5]
```diff
@@ -20,9 +22,30 @@ def deps do
...
+
+```bash
+pacman -S ffmpeg
+```
+
+#### MacOS
+
+```bash
+brew install ffmpeg
+```
```
Comment (Rados13) [inline README.md]: Shouldn't we simply refer to ffmpeg website with instructions for installing it instead of providing commands for some of the OS.

**Review by Rados13 — APPROVED** [review]

**Pull request merged**